mempool/src/shared_mempool.rs

```diff
@@ -449,65 +448,66 @@ where
     crit!("SharedMempool outbound_sync_task terminated");
 }
 
-async fn process_consensus_request<V>(
-    smp: SharedMempool<V>,
-    msg: MempoolRequest,
-    callback: oneshot::Sender<Result<MempoolResponse>>,
-) where
+async fn process_state_sync_request<V>(smp: SharedMempool<V>, req: CommitNotification)
+where
     V: TransactionValidation,
 {
-    let resp = match msg {
-        MempoolRequest::GetBlockRequest {
-            max_block_size,
-            transactions,
-        } => {
-            let block_size = cmp::max(max_block_size, 1);
-            counters::MEMPOOL_SERVICE
-                .with_label_values(&["get_block", "requested"])
-                .inc_by(block_size as i64);
-
-            let exclude_transactions: HashSet<TxnPointer> = transactions
-                .iter()
-                .map(|txn| (txn.sender, txn.sequence_number))
-                .collect();
-
-            let mut txns = smp
-                .mempool
-                .lock()
-                .expect("[get_block] acquire mempool lock")
-                .get_block(block_size, exclude_transactions);
-
-            let transactions = txns.drain(..).map(SignedTransaction::into).collect();
+    {
+        let mut pool = smp
+            .mempool
+            .lock()
+            .expect("[shared mempool] failed to get mempool lock");
 
-            MempoolResponse::GetBlockResponse { transactions }
+        for transaction in req.transactions {
+            pool.remove_transaction(&transaction.sender, transaction.sequence_number, false);
         }
-        MempoolRequest::CommitTransactionsRequest {
-            transactions,
-            block_timestamp_usecs,
-        } => {
-            let mut pool = smp.mempool.lock().unwrap();
-            for transaction in transactions.iter() {
-                pool.remove_transaction(
-                    &transaction.sender,
-                    transaction.sequence_number,
-                    transaction.is_rejected,
-                );
-            }
+        if req.block_timestamp_usecs > 0 {
+            pool.gc_by_expiration_time(Duration::from_micros(req.block_timestamp_usecs));
+        }
+    }
 
-            if block_timestamp_usecs > 0 {
-                pool.gc_by_expiration_time(Duration::from_micros(block_timestamp_usecs));
-            }
+    let resp = CommitResponse {};
```

Comment:
I added it because in old implementation consensus did get ACK back from notifying mempool about committed txns. I guess tradeoff here is simplicity vs. feedback for consensus, but it's not like it can actually act on that info other than logging.

Reply:
yeah.....we talked offline. Looks like we want to keep flow serialized. So it's necessary